tests/e2e/pubsub/pubsub_test.go

```diff
@@ -316,6 +450,17 @@ func TestMain(m *testing.M) {
 			AppMemoryLimit:   "200Mi",
 			AppMemoryRequest: "100Mi",
 		},
+		{
```

Comment:
I would split it into separate test just for routing. Same for gRPC. I think it might be better to maintain those tests independently.

Reply:
Agreed. I'm going to leave some of the cloud event helper code in case that is helpful in the future.